lib/src/core/optional.dart

```diff
@@ -89,7 +89,7 @@ class Optional<T> extends IterableBase<T> {
   /// The transformer must not return [null]. If it does, an [ArgumentError] is thrown.
   Optional<S> transform<S>(S transformer(T value)) {
     return _value == null
-        ? const Optional.absent()
+        ? Optional<S>.absent()
```

Comment:
One way to avoid the extra allocation is to keep a static `Map<Type, Optional>` cache. Since there's an upper bound for the number of types in any given Dart program this map won't grow indefinitely. Of course, that adds a map look-up.

Reply:
It adds a map lookup and a static cache. That said this just reverts back to the same number of allocations we did before e769045.